pkg/execution/executor/executor.go

```diff
@@ -659,47 +747,34 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 			}
 		} else if resp.IsTraceVisibleFunctionExecution() {
 			spanName := "function success"
+			fnstatus := attribute.String(consts.OtelSysFunctionStatus, enums.RunStatusCompleted.String())
+
 			if resp.StatusCode != 200 {
 				spanName = "function error"
+				fnstatus = attribute.String(consts.OtelSysFunctionStatus, enums.RunStatusFailed.String())
 				span.SetStatus(codes.Error, resp.Error())
 			}
 
-			span.SetName(spanName)
+			fnSpan.SetAttributes(fnstatus)
 
+			span.SetName(spanName)
 			span.SetAttributes(
-				attribute.Int(consts.OtelSysFunctionStatus, resp.StatusCode),
+				attribute.Int(consts.OtelSysFunctionStatusCode, resp.StatusCode),
 			)
 
 			if byt, err := json.Marshal(resp.Output); err == nil {
 				span.AddEvent(string(byt), trace.WithAttributes(
 					attribute.Bool(consts.OtelSysFunctionOutput, true),
 				))
 			}
-
 		} else {
-			// Only add this span if it's a step or function response that
-			// represents either a failed or a successful execution. Do not
-			// record discovery spans.
-			span.SetAttributes(
-				attribute.Bool(consts.OtelSysIgnored, true),
-			)
+			// if it's not a step or function response that represents either a failed or a successful execution.
+			// Do not record discovery spans and cancel it.
+			ctx = span.Cancel(ctx)
 		}
 	}
 
 	err = e.HandleResponse(ctx, id, item, edge, resp)
-
-	if err == nil && item.Kind == queue.KindStart {
-		md := s.Metadata()
-		// This is the first successful step, so mark the fn as started.
-		_ = e.sm.UpdateMetadata(ctx, s.RunID(), state.MetadataUpdate{
-			Debugger:                  md.Debugger,
-			Context:                   md.Context,
-			DisableImmediateExecution: md.DisableImmediateExecution,
-			RequestVersion:            md.RequestVersion,
-			StartedAt:                 start,
```

Comment:
move this metadata update here instead since we need it to have a consistent start time.
https://github.com/inngest/inngest/pull/1248/files#diff-ccd0679a9f24c84fef4f5d4d5fe2c32651d23886d85af8448bb968921c7b8a95R687-R698